source/ceylon/future/Future.ceylon

```diff
@@ -0,0 +1,92 @@
+import ceylon.future.util {
+    Latch,
+    AcquireTimeoutException
+}
+import ceylon.promise {
+    Promise,
+    ExecutionContext
+}
+
+"Wraps a [[Promise]] that will be completed by another thread."
```

Comment:
Actually it says that will be completed by another thread. That's actually what promise do (they are completed by other thread usually or the same thread in an event loop based system). Anyway, the point of the future is to provide a _blocking_ API for obtaining the promised value.

So I would rather say something like "Extend a promise and provide a blocking API for obtaining the resolvable value".